ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java

```diff
@@ -53,7 +55,8 @@
   private final BlockImporter blockImporter;
   private final ChainStorageClient chainStorageClient;
   private final ArtemisConfiguration config;
-  private final List<DepositWithIndex> deposits = new ArrayList<>();
+  private final NavigableSet<DepositWithIndex> deposits = new TreeSet<>();
+  private Bytes32 genesisEth1BlockHash;
```

Comment:
Should this be `volatile`?

Reply:
Alternatively, could we just make this a local variable within `onDeposit` and always set it to the last deposit's blockHash?

Reply:
Doesn't matter now, but each `@Subscribe` method is called inside a `synchronized` block (specific to the method, not the class) so as long as the variables are only used as a result of a single event method, we can treat it as single threaded.